pkg/metadata/host/host_windows.go

```diff
@@ -5,14 +5,162 @@
+	"github.com/shirou/w32"
+)
+
+var (
+	modkernel = syscall.NewLazyDLL("kernel32.dll")
+
+	procGetTickCount64 = modkernel.NewProc("GetTickCount64")
+)
+
+// InfoStat describes the host status.  This is not in the psutil but it useful.
+type InfoStat struct {
+	Hostname             string `json:"hostname"`
+	Uptime               uint64 `json:"uptime"`
+	BootTime             uint64 `json:"bootTime"`
+	Procs                uint64 `json:"procs"`           // number of processes
+	OS                   string `json:"os"`              // ex: freebsd, linux
+	Platform             string `json:"platform"`        // ex: ubuntu, linuxmint
+	PlatformFamily       string `json:"platformFamily"`  // ex: debian, rhel
+	PlatformVersion      string `json:"platformVersion"` // version of the complete OS
+	KernelVersion        string `json:"kernelVersion"`   // version of the OS kernel (if available)
+	VirtualizationSystem string `json:"virtualizationSystem"`
+	VirtualizationRole   string `json:"virtualizationRole"` // guest or host
+	HostID               string `json:"hostid"`             // ex: uuid
+}
 type osVersion [2]string
 
 //Set the OS to "win32" instead of the runtime.GOOS of "windows" for the in app icon
 const osName = "win32"
 
-func fillOsVersion(stats *systemStats, info *host.InfoStat) {
+// Collect at init time
+var cpuInfo []InfoStat
+
+// InitHostMetadata initializes necessary CPU info
+func InitHostMetadata() error {
+	var err error
+	info := getHostInfo()
+	cpuInfo = append(cpuInfo, *info)
+
+	return err
+
+}
+
+func fillOsVersion(stats *systemStats, info *InfoStat) {
 	// TODO
 	stats.Winver = osVersion{info.PlatformFamily, info.PlatformVersion}
 }
+
+// GetStatusInformation just returns an InfoStat object, we need some additional information that's not
+func GetStatusInformation() *InfoStat {
+	return getHostInfo()
+}
+
+func getSystemStats() *systemStats {
+	var stats *systemStats
+	key := buildKey("systemStats")
+	if x, found := cache.Cache.Get(key); found {
+		stats = x.(*systemStats)
+	} else {
+		cpuInfo, _ := cpu.GetCpuInfo()
+		hostInfo := getHostInfo()
+		cores, _ := strconv.Atoi(cpuInfo["cpu_cores"])
+		c32 := int32(cores)
+
+		stats = &systemStats{
+			Machine:   runtime.GOARCH,
+			Platform:  runtime.GOOS,
+			Processor: cpuInfo["model_name"],
+			CPUCores:  c32,
+			Pythonv:   strings.Split(getPythonVersion(), " ")[0],
+		}
+
+		// fill the platform dependent bits of info
+		fillOsVersion(stats, hostInfo)
+		cache.Cache.Set(key, stats, cache.NoExpiration)
+	}
+
+	return stats
+}
+
+func getHostInfo() *InfoStat {
+	key := buildKey("hostInfo")
+	if x, found := cache.Cache.Get(key); found {
+		return x.(*InfoStat)
+	}
+
+	info := &InfoStat{}
+	info.Hostname, _ = os.Hostname()
+
+	upTime := time.Duration(getTickCount64()) * time.Millisecond
+	bootTime := time.Now().Add(-upTime)
+	info.Uptime = uint64(upTime.Seconds())
+	info.BootTime = uint64(bootTime.Unix())
+	pids, _ := Pids()
+	info.Procs = uint64(len(pids))
```

Comment:
Do we do this for linux ? If not: is it really useful ?

Reply:
we do it on linux, too https://github.com/shirou/gopsutil/blob/master/host/host_linux.go#L71-L73